src/main/java/mezz/jei/util/StringUtil.java

```diff
@@ -1,13 +1,15 @@
 package mezz.jei.util;
 
 import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.StringTextComponent;
 
 public final class StringUtil {
 	private StringUtil() {
 
 	}
 
-	public static String truncateStringToWidth(String string, int width, FontRenderer fontRenderer) {
-		return fontRenderer.getLineScrollOffset(string, width - fontRenderer.getStringWidth("..."), false) + "...";
+	public static ITextComponent truncateStringToWidth(ITextComponent text, int width, FontRenderer fontRenderer) {
+		return new StringTextComponent(fontRenderer.func_238417_a_(text, width - fontRenderer.getStringWidth("...")).getString() + "...");
```

Comment:
would be nice to introduce some intermediate variables here, I try to avoid having more than one function call per line